src/lib/i18n.jsx

```diff
@@ -72,11 +73,44 @@ export function useLanguage() {
 }
 
 /**
- * Get the translation of a key. Defaults to english if no translation is found
+ * Get the translation of a key. Defaults to English if no translation is found
  * @param {string} key
  */
 export function useTranslation(key) {
 	const [lang] = useLanguage();
 	const data = config.i18n[key];
 	return data[lang] || data.en;
 }
+
+/**
+ * Get the translated name of a path based upon the current language.
+ * @param {string} path
+ */
+export function useNavTranslation(path) {
+	const [lang] = useLanguage();
+
+	for (const route in config.nav) {
+		if (config.nav[route].path === path) {
+			return getRouteName(config.nav[route], lang);
+		} else if (config.nav[route].routes) {
+			for (const subRoute of config.nav[route].routes) {
+				if (subRoute.path === path) {
+					return getRouteName(subRoute, lang);
+				}
+			}
+		}
+	}
+
+	return null;
+}
+
+/**
+ * @param {{ name: Record<string, string> | string }} route
+ * @param {string} lang
+ * @return {string}
+ */
+export function getRouteName(route, lang) {
+	return typeof route.name === 'object'
+		? route.name[lang] || route.name.en
+		: route.name;
+}
```

Comment:
More of a hold over mechanism, I hope to switch the nav translations into a keyed object & break them away from the route paths in the future.